### ISSUE TITLE
Numerical test for LedoitWolf

### ISSUE THREAD

**Issue by tlienart** [mateuszbaran/CovarianceEstimation.jl#14]
Here's the example they use in R (seeded and written to file)

```R
set.seed(1234)
library("nlshrink")
# generate matrix of uniform random variates
X <- matrix(sapply(1:20, function(b) runif(50, max=b)), nrow = 50, ncol = 20)
Sigma <- diag((1:20)^2/12) # true population covariance matrix
nlshrink_X <- nlshrink_cov(X, k=0) # compute non-linear shrinkage estimate
linshrink_X <- linshrink_cov(X, k=0) # compute linear shrinkage estimate
S <- cov(X) # sample covariance matrix
# compare accuracy of estimators (sum of squared elementwise Euclidean distance)
sum((S-Sigma)^2) # 1459.078
sum((nlshrink_X - Sigma)^2) # 670.6714
sum((linshrink_X - Sigma)^2) # 781.236
write.table(X, file="mat.dat", row.names=FALSE, col.names=FALSE)
```

(if you don't have R handy: https://github.com/tlienart/Dump/blob/master/jl/covarianceestimation/test-mat.dat) 

in Julia

```julia
using DelimiterFiles, CovarianceEstimation, LinearAlgebra
X = readdlm("test-mat.dat");
Σ = diagm(0=>[i^2/12 for i ∈ 1:20])
C = cov(X);
CLW = cov(LedoitWolf(), X);
CC = cov(RaoBlackwellLedoitWolf(), X);
sum((C-Σ).^2) # 1459.078 -- sanity check
sum((CLW-Σ).^2) # 254.399
sum((CC-Σ).^2) # 973.95
```

so I guess LW makes sense here, not sure about RBLW though it's not off the charts. I was assuming their `nlshrink_cov` to be our `LedoitWolf` but it doesn't look that way (though here it's in our favour?)
It looks like they're doing something quite different though (computing eigenvalues and the QuEST function?), their evaluation takes roughly 1.5s on my machine whilst ours takes 230μs... 

It'd be good to have something of the sorts where we're sure of the results and use this as a test?

**Comment by tlienart** [comment]
I also tried `corpcor` which I believe uses another shrinkage method. 

```R
library("corpcor")
scc = cov.shrink(X)
sum((scc-Sigma)^2) # 214.2
```

Which is on par with the `254.399` above. That function is faster, takes about 2.5ms, still an order of magnitude slower than our LW which is nice I guess.

**Comment by mateuszbaran** [comment]
Nice analysis 👍 . `nlshrink_cov`, as far as I understand, implements a different estimator. I haven't yet read the papers they cite but I've seen another paper by Ledoit and Wolf where they describe a nonlinear shrinkage estimation method. `linshrink_cov` should be, more or less, what we have here but it's impossible to be sure it 100% conforms to the paper without reading the source code. corpcor's `shrink` implements yet another estimator as indicated in their documentation. It is still nice that we have a fast implementation that returns approximately correct results.

The covariance matrix in LW should indeed be uncorrected.

And thanks for that link to reference implementations, it is valuable.

Transposing `X` is a not-yet-corrected effect of the fact that the code was originally designed with MultivariateStats.jl's LDA in mind.

**Comment by tlienart** [comment]
now in #15 , can be closed upon merging.

**Comment by mateuszbaran** [comment]
Fixed by PR #15
